packages/IntegrationTest/windows/integrationtest/TestHostHarness.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "TestHostHarness.h"
+#include "TestModule.h"
+
+#include <JSValue.h>
+
+using namespace winrt::integrationtest;
+using namespace winrt::Microsoft::ReactNative;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Data::Json;
+
+namespace IntegrationTest {
+
+TestHostHarness::TestHostHarness() noexcept {
+  m_redboxHandler = winrt::make<TestHostHarnessRedboxHandler>(get_weak());
+  m_commandListener = winrt::make_self<TestCommandListener>();
+
+  m_commandListener->OnTestCommand(
+      [weakThis{get_weak()}](const TestCommand &command, TestCommandResponse response) noexcept {
+        if (auto strongThis = weakThis.get()) {
+          strongThis->OnTestCommand(command, std::move(response));
+        }
+      });
+}
+
+void TestHostHarness::SetRootView(ReactRootView &&rootView) noexcept {
+  m_rootView = std::move(rootView);
+}
+
+void TestHostHarness::SetReactHost(ReactNativeHost &&reactHost) noexcept {
+  reactHost.InstanceSettings().RedBoxHandler(m_redboxHandler);
+  m_instanceLoadedRevoker = reactHost.InstanceSettings().InstanceLoaded(
+      winrt::auto_revoke, [weakThis{get_weak()}](const auto & /*sender*/, const auto &args) noexcept {
+        if (auto strongThis = weakThis.get()) {
+          strongThis->m_context = args.Context();
+          strongThis->StartListening();
+        }
+
+        ReactNotificationService notifications(args.Context().Notifications());
```

Comment:
Should consider wrapping the `IReactContext` in a `ReactContext` earlier to not need to deal with the IInspectable vs Wrapper issues.

Reply:
Kind of a question for @vmoroz, but it ends up feeling kind of weird to add an event handler to the non-ABI `ReactNativeHost`, then get back ABI `InstanceLoadedEventArgs` in the callback. Not being super familiar with the mechanics of this, it ended up being kind of confusing to need to rewrap. Is there any sort of clean way to have `ReactInstanceSettings` lifetime event args use the non-ABI `ReactContext`?